cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_data_adapters.go

```diff
@@ -400,33 +411,207 @@ func (s *Server) buildPkgInstall(installedPackageName, targetCluster, targetName
 	return pkgInstall, nil
 }
 
-func getPackageRepository(pr *packagingv1alpha1.PackageRepository) (*v1alpha1.PackageRepository, error) {
-	// See the PackageRepository CR at
-	// https://carvel.dev/kapp-controller/docs/latest/packaging/#packagerepository-cr
+// package repositories
+
+func (s *Server) buildPackageRepositorySummary(pr *packagingv1alpha1.PackageRepository, cluster string) (*corev1.PackageRepositorySummary, error) {
+
+	// base struct
+	repository := &corev1.PackageRepositorySummary{
+		PackageRepoRef: &corev1.PackageRepositoryReference{
+			Context: &corev1.Context{
+				Cluster:   cluster,
+				Namespace: pr.Namespace,
+			},
+			Plugin:     GetPluginDetail(),
+			Identifier: pr.Name,
+		},
+		Name:            pr.Name,
+		NamespaceScoped: s.globalPackagingNamespace != pr.Namespace,
+	}
+
+	// handle fetch-specific configuration
```

Comment:
sorry this is confusing a bit. In create you have code to error if type != image bundle. 
here you have code to handle all kinds?

Reply:
I thought to make a difference between reading vs. writing.
if the repository exists in the cluster, there is no reason we cannot view it (at least part of it) and at least know of it existence.
but for creation/update, we cannot provide enough details, so creation/update is not supported.

Reply:
maybe add a comment to that effect in the code?